src/dict.h

```diff
@@ -196,6 +196,7 @@ int dictTryExpand(dict *d, unsigned long size);
 int dictShrink(dict *d, unsigned long size);
 int dictAdd(dict *d, void *key, void *val);
 dictEntry *dictAddRaw(dict *d, void *key, dictEntry **existing);
+dictEntry *dictAddNonExistingRaw(dict *d, void *key, const uint64_t *hash);
```

Comment:
Assuming you agree about replacing `dictFindWithHash` to `dictFindByHash`. Maybe rename to `dictAddNonExistsByHash()` ? The name suggest to the coupling among the two.

Reply:
addressed in 0a111c1458617403e02cd385ca31edcaeb0516eb